internal/jimm/ssh/ssh.go

```diff
@@ -51,25 +51,50 @@ type SSHKeyManager interface {
 	VerifyPublicKey(ctx context.Context, claimUser string, publicKey []byte) (bool, error)
 }
 
+// SSHDialer provides a means to establish an SSH connection.
+type SSHDialer interface {
+	Dial(network string, addr string, config *gossh.ClientConfig) (*gossh.Client, error)
+}
+
+// BasicDialer is a wrapper around the default Go x/crypto/ssh
+// dialer for cases where no changes are needed.
+type BasicDialer struct{}
+
+func (d *BasicDialer) Dial(network string, addr string, config *gossh.ClientConfig) (*gossh.Client, error) {
+	return gossh.Dial(network, addr, config)
+}
+
+type SSHManagerParams struct {
+	IdentityManager IdentityManager
+	JujuManager     JujuManager
+	SSHKeyManager   SSHKeyManager
+	JWTFactory      *jujuauth.Factory
+	Dialer          SSHDialer
+}
+
 // NewSSHManager returns a new SSHManager that offers jimm functionality to the SSHJumpServer.
-func NewSSHManager(identityManager IdentityManager, jujuManager JujuManager, sshKeyManager SSHKeyManager, jwtFactory *jujuauth.Factory) (*sshManager, error) {
-	if identityManager == nil {
+func NewSSHManager(p SSHManagerParams) (*sshManager, error) {
+	if p.IdentityManager == nil {
 		return nil, errors.E("identityManager cannot be nil")
 	}
-	if jujuManager == nil {
+	if p.JujuManager == nil {
 		return nil, errors.E("jujuManager cannot be nil")
 	}
-	if sshKeyManager == nil {
+	if p.SSHKeyManager == nil {
 		return nil, errors.E("sshManager cannot be nil")
 	}
-	if jwtFactory == nil {
+	if p.JWTFactory == nil {
 		return nil, errors.E("jwtFactory cannot be nil")
 	}
+	if p.Dialer == nil {
+		return nil, errors.E("dialer cannot be nil")
+	}
 	return &sshManager{
-		jujuManager:     jujuManager,
-		identityManager: identityManager,
-		sshKeyManager:   sshKeyManager,
-		jwtFactory:      jwtFactory,
+		jujuManager:     p.JujuManager,
```

Comment:
seems like sshManager could just embed the SSHManagerParams?

Reply:
True, I'll do that

Reply:
Ah, it can't right now, because the params are all exported, while the types on the actual manager are not exported. That could change in the future but I won't do it in this PR.